src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -66,6 +67,8 @@ public partial class QuicListenerOptions : System.Net.Quic.QuicOptions
         public int ListenBacklog { get { throw null; } set { } }
         public System.Net.IPEndPoint? ListenEndPoint { get { throw null; } set { } }
         public System.Net.Security.SslServerAuthenticationOptions? ServerAuthenticationOptions { get { throw null; } set { } }
+        public System.Net.Quic.QuicListener.ServerOptionsSelectionCallback? ServerOptionsSelectionCallback { get { throw null; } set { } }
+        public object? ServerOptionsSelectionCallbackState { get { throw null; } set { } }
```

Comment:
Do we generally let you specify a state object for callbacks like this? We don't do that for most callbacks that I'm aware of.

Reply:
We did for SslStream: https://docs.microsoft.com/en-us/dotnet/api/system.net.security.serveroptionsselectioncallback?view=net-5.0
The goal I was trying to achieve was to make HTTP/3 as similar to previous versions as possible. 
I think the motivation was to make it easier for the callback logic to figure out what to do. 

However I'm not sure if the Listener is same as there is no concept of Listener on SslStream. There is only the 
[AuthenticateAsServerAsync](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasserverasync?view=net-5.0#system-net-security-sslstream-authenticateasserverasync)

cc: @davidfowl who originally asked for the state.

Reply:
gone.